### PR TITLE
Enable _dev/deploy for data streams

### PR DESCRIPTION
**Pull request by mtojek**
Issue: https://github.com/elastic/elastic-package/issues/89

This PR enables custom `_dev/deploy` for data stream. It will be used by the future AWS service deployer.

### CODE REVIEW

**Comment by ycombinator** [comment]
I think this PR depends on https://github.com/elastic/package-spec/pull/111? I will review this PR once https://github.com/elastic/package-spec/pull/111 is reviewed and merged.

**Comment by mtojek** [comment]
I updated the package-spec in this branch.

File: internal/testrunner/runners/system/servicedeployer/factory.go [hunk 1]
```diff
@@ -5,27 +5,62 @@
...
+
+// Factory chooses the appropriate service runner for the given data stream, depending
+// on service configuration files defined in the package or data stream.
+func Factory(options FactoryOptions) (ServiceDeployer, error) {
+	devDeployPath, err := findDevDeployPath(options)
+	if err != nil {
+		logger.Errorf("can't find _dev/deploy directory")
+		return nil, ErrNotFound
+	}
+	fmt.Println(devDeployPath)
```
Comment (ycombinator) [inline internal/testrunner/runners/system/servicedeployer/factory.go:39]: Should be removed?

File: internal/testrunner/runners/system/servicedeployer/factory.go [hunk 2]
```diff
@@ -5,27 +5,62 @@
...
+	} else if !os.IsNotExist(err) {
+		return "", errors.Wrapf(err, "stat failed (path: %s)", dataStreamDevDeployPath)
+	}
+
+	packageDevDeployPath := filepath.Join(options.PackageRootPath, devDeployDir)
+	_, err = os.Stat(packageDevDeployPath)
+	if err == nil {
+		return packageDevDeployPath, nil
+	} else if !os.IsNotExist(err) {
+		return "", errors.Wrapf(err, "stat failed (path: %s)", packageDevDeployPath)
```
Comment (ycombinator) [inline internal/testrunner/runners/system/servicedeployer/factory.go:63]: Both the errors (the one and the one above for the data stream) are very similar. Can you differentiate them a bit by saying that one failed for the data stream and one for the package?

File: internal/testrunner/runners/system/servicedeployer/factory.go (hunk 1 above)
Comment (mtojek) [reply internal/testrunner/runners/system/servicedeployer/factory.go]: Fixed, thanks!

File: internal/testrunner/runners/system/servicedeployer/factory.go (hunk 2 above)
Comment (mtojek) [reply internal/testrunner/runners/system/servicedeployer/factory.go]: The path argument would be different, but I extended the error message to include also some context.

**Review by ycombinator — APPROVED** [review]
LGTM.

**Pull request merged**